pyspecdata/core.py

```diff
@@ -6887,132 +6900,55 @@ def __init__(self,*args,**kwargs):
-        # for now, I'm going to assume that it's not using sigma, though this could be wrong
-        # and I could need to scale everything by sigma in the same way as errfunc
-        return self.parameter_derivatives(x,set = self.symbol_list,set_to = p).T
-    def analytical_covariance(self):
-        covarmatrix = zeros([len(self._active_symbols())]*2)
-        #{{{ try this ppt suggestion --> his V is my fprime, but 
-        fprime = self.parameter_derivatives(self.getaxis(self.fit_axis))
-        dirproductform = False
-        if dirproductform:
-            sigma = self.get_error()
-            f1 = fprime.shape[0]
-            f2 = fprime.shape[1]
-            fprime1 = fprime.reshape(f1,1,f2) # j index
-            fprime2 = fprime.reshape(1,f1,f2) # k index
-            fprime_prod = fprime1 * fprime2
-            fprime_prod = fprime_prod.reshape(-1,f2).T # direct product form
-            try:
-                covarmat = dot(pinv(fprime_prod),(sigma**2).reshape(-1,1))
-            except ValueError as e:
-                raise ValueError(strm('shape of fprime_prod', shape(fprime_prod),
-                    'shape of inverse', shape(pinv(fprime_prod)),
-                    'shape of sigma', shape(sigma))+explain_error(e))
-            covarmatrix = covarmat.reshape(f1,f1)
-            for l in range(0,f1): 
-                for m in range(0,f1): 
-                    if l != m:
-                        covarmatrix[l,m] /= 2
-        else:
-            sigma = self.get_error()
-            #covarmatrix = dot(pinv(f),
-            #        dot(diag(sigma**2),pinv(f.T)))
-            J = matrix(fprime.T)
-            #W = matrix(diag(1./sigma**2))
-            #S = matrix(diag(sigma**2))
-            #if hasattr(self,'data_covariance'):
-            #    print "covariance data is present"
-            S = matrix(self.get_covariance())
-            Omegainv = S**-1
-            #S = matrix(diag(sigma**2))
-            #G = matrix(diag(1./sigma))
-            #G = S**(-1/2) # analog of the above
-            #covarmatrix = ((J.T * W * J)**-1) * J.T * W
-            print('a')
-            minimizer = inv(J.T * Omegainv * J) * J.T * Omegainv
-            covarmatrix = minimizer * S * minimizer.T
-            #covarmatrix = array(covarmatrix * S * covarmatrix.T)
-            #covarmatrix = array((J.T * G.T * G * J)**-1 * J.T * G.T * G * S * G.T * G * J * (J.T * G.T * G * J)**-1)
-            #try:
-            #    betapremult = (J.T * Omegainv * J)**-1 * J.T * Omegainv
-            #except:
-            #    print 'from sigma','\n\n',diag(sigma**2),'\n\n','from covarmatrix','\n\n',S,'\n\n'
-            #    raise RuntimeError('problem generating estimator (word?)')
-            #covarmatrix = array( betapremult * S * betapremult.T)
-        #print "shape of fprime",shape(fprime),"shape of fprime_prod",shape(fprime_prod),'sigma = ',sigma,'covarmat=',covarmatrix,'\n'
+    @property
+    def function_string(self):
```

Comment:
I would add a docstring here.. I would say the docstring isn't required for the setter, etc, but should be present here

Reply:
Ok, I have added this